test/test_autograd.py

```diff
@@ -9281,17 +9281,48 @@ def run_test(fn):
         run_test(lambda x: x.add_(2))
         run_test(lambda x: x.transpose_(0, 1))
 
+
 class TestMultithreadAutograd(TestCase):
     def _run_py_multithread_fn(self, fn, args=(), num_threads=10, kwargs=None):
+
+        class PropagatingThread(threading.Thread):
+            '''Helper class to propagate exception from child
+            thread to main thread on join.
+
+            Reference: https://stackoverflow.com/a/31614591/5602957
+            '''
+
+            def run(self):
+                self.exception = None
+                try:
```

Comment:
Do we need to worry about refcycles here?
cf. https://github.com/python/cpython/blob/3.9/Lib/threading.py#L912-L914
Maybe we can call the parent method instead?